containers/src/Kokkos_DynamicView.hpp

```diff
@@ -53,36 +53,184 @@
 namespace Kokkos {
 namespace Experimental {
 
-// Simple metafunction for choosing memory space
-// In the current implementation, if memory_space == CudaSpace,
-// use CudaUVMSpace for the chunk 'array' allocation, which
-// contains will contain pointers to chunks of memory allocated
-// in CudaSpace
 namespace Impl {
-template <class MemSpace>
-struct ChunkArraySpace {
-  using memory_space = MemSpace;
-};
 
-#ifdef KOKKOS_ENABLE_CUDA
-template <>
-struct ChunkArraySpace<Kokkos::CudaSpace> {
-  using memory_space = typename Kokkos::CudaUVMSpace;
-};
-#endif
-#ifdef KOKKOS_ENABLE_HIP
-template <>
-struct ChunkArraySpace<Kokkos::Experimental::HIPSpace> {
-  using memory_space = typename Kokkos::Experimental::HIPHostPinnedSpace;
+template <typename MemorySpace, typename ValueType>
+struct Destroy {
```

Comment:
I think this is not a good name for something Kokkos::Impl:: if its DynamicView specific. Either put it into an Anonymous namespace, make it a nested class in DynamicView, or rename to something DynamicView related. I just think its easy to have naming conflicts with other stuff in the Kokkos namespace since its so generic.

Reply:
Also a short comment for this class what it is responsible for would be good for documentation purposes (though its fairly obvious in this case)

Reply:
Ah just noticed that it was named that way before. So won't block on it but its still a terrible name :-D

Reply:
Good idea, I will improve the name!

Reply:
To solve this, I moved it to be a private inner class of `SmartMemoryAccessor`, but the left the name as is now that it's not accessible outside of that inner class.